python/ray/experimental/sgd/tensorflow/tensorflow_runner.py

```diff
@@ -0,0 +1,98 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import logging
+
+import ray
+from ray.experimental.sgd import utils
+
+logger = logging.getLogger(__name__)
+
+
+class TensorFlowRunner(object):
+    """Manages a TensorFlow model for training."""
+
+    def __init__(self, model_creator, data_creator, config=None,
+                 batch_size=16):
+        """Initializes the runner.
+
+        Args:
+            model_creator (dict -> Model): see tensorflow_trainer.py.
+            data_creator (dict -> BatchDataset, BatchDataset):
+                see tensorflow_trainer.py.
+            config (dict): see tensorflow_trainer.py.
+            batch_size (int): see tensorflow_trainer.py.
+        """
+
+        self.model_creator = model_creator
+        self.data_creator = data_creator
+        self.config = {} if config is None else config
+        self.batch_size = batch_size
+        self.verbose = True
+
+        self.epoch = 0
+
+    def setup(self):
+        """Initializes the model."""
+        logger.debug("Creating model")
+        self.model = self.model_creator()
+
+        logger.debug("Creating dataset")
+        self.train_dataset, self.test_dataset = self.data_creator(
+            self.batch_size)
+
+    def step(self):
+        """Runs a training epoch and updates the model parameters."""
+
+        history = self.model.fit(self.train_dataset, verbose=0)
+
+        stats = {"train_loss": history.history["loss"][-1]}
+
+        self.epoch += 1
+        return stats
+
+    def validate(self):
+        """Evaluates the model on the validation data set."""
+        stats = {}
+        results = self.model.evaluate(self.train_dataset, verbose=0)
+        stats["train_loss"] = results[0]
+        results = self.model.evaluate(self.test_dataset, verbose=0)
+        stats["validation_loss"] = results[0]
+        return stats
+
+    def get_state(self):
+
+        return {
+            "epoch": self.epoch,
+            "weights": self.model.get_weights(),
+            "optimizer_weights": self.model.optimizer.get_weights()
+        }
+
+    def set_state(self, state):
+        self.epoch = state["epoch"]
+        if self.model.optimizer.weights == []:
```

Comment:
I don't quite understand this issue nor this comment - can you:

1. provide a comment of exactly what the error is, and
2. provide in the code a link to the stackoverflow or Tensorflow github issue link that you found which suggested this workaround?